src/platform_impl/macos/mod.rs

```diff
@@ -47,20 +47,42 @@ extern "C" {
 #[allow(non_upper_case_globals)]
 const NSAboutPanelOptionCopyright: &str = "Copyright";
 
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct Menu {
     id: u32,
     ns_menu: id,
     children: Rc<RefCell<Vec<Rc<RefCell<MenuChild>>>>>,
 }
 
+impl Clone for Menu {
+    fn clone(&self) -> Self {
+        unsafe {
+            let _: () = msg_send![self.ns_menu, retain];
+        }
+        Self {
+            id: self.id,
+            ns_menu: self.ns_menu.clone(),
+            children: self.children.clone(),
+        }
+    }
+}
+
```

Comment:
we can remove this, we don't need it anymore, Clone is implemented for the outer type.
```suggestion
```